.github/workflows/operator-integration-test.yml

```diff
@@ -110,3 +110,202 @@ jobs:
           cd integration-tests/manifests/cmd
           go run validate_instrumentation_vars.go default ns_instrumentation_env_variables.json
           kubectl delete instrumentation sample-instrumentation
+
+
+
+  DeploymentAnnotationsTest:
+    name: DeploymentAnnotationsTest
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Start minikube
+        uses: medyagh/setup-minikube@master
+
+      - name: Deploy cert-manager to minikube
+        run:
+          kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.12.0/cert-manager.yaml
+
+      - name: Verify minikube and cert-manager
+        run: |
+          sleep 10
+          kubectl get pods -A
+
+      - name: Build image
+        run: |
+          eval $(minikube docker-env)
+          make container
+          docker images
+
+      - name: Deploy operator to minikube
+        run: |
+          make deploy
+
+      - name: Test Annotations
+        run: |
+          kubectl apply -f integration-tests/manifests/sample-deployment.yaml
+          kubectl get pods -A
+          kubectl describe pods -n default
+          sleep 10
+          go test -v -run TestJavaAndPythonDeployment ./integration-tests/manifests/annotations -timeout 30m
+          sleep 10
+          go test -v -run TestJavaOnlyDeployment ./integration-tests/manifests/annotations -timeout 30m
+          sleep 10
+          go test -v -run TestPythonOnlyDeployment ./integration-tests/manifests/annotations -timeout 30m
+          kubectl get pods -A
+          kubectl describe pods -n default
+
+ 
+
+  DaemonsetAnnotationsTest:
+    name: DaemonsetAnnotationsTest
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Start minikube
```

Comment:
nit: you can put steps "Start minikube" to "Deploy operator to minikube" as a reusable workflow and call it in all the tests. https://www.dhiwise.com/post/the-ultimate-guide-to-github-reusable-workflows-maximize-efficiency-and-collaboration